src/backends/torch/torchlib.cc

```diff
@@ -0,0 +1,142 @@
+ */
+
+#include "torchlib.h"
+
+#include <torch/script.h>
+
+#include "outputconnectorstrategy.h"
+
+using namespace torch;
+
+namespace dd
+{
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::TorchLib(const TorchModel &tmodel)
+        : MLLib<TInputConnectorStrategy,TOutputConnectorStrategy,TorchModel>(tmodel) 
+    {
+        this->_libname = "torch";
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::TorchLib(TorchLib &&tl) noexcept
+        : MLLib<TInputConnectorStrategy,TOutputConnectorStrategy,TorchModel>(std::move(tl))
+    {
+        this->_libname = "torch";
+        _traced = std::move(tl._traced);
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::~TorchLib() 
+    {
+        
+    }
+
+    /*- from mllib -*/
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    void TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::init_mllib(const APIData &ad) 
+    {
+        bool gpu = false;
+        APIData lib_ad = ad.getobj("parameters").getobj("mllib");
+
+        if (lib_ad.has("gpu")) {
+            gpu = lib_ad.get("gpu").get<bool>() && torch::cuda::is_available();
+        }
+        if (lib_ad.has("nclasses")) {
+            _nclasses = lib_ad.get("nclasses").get<int>();
+        }
+
+        _device = gpu ? torch::Device("cuda") : torch::Device("cpu");
+
+        _traced = torch::jit::load(this->_mlmodel._model_file);
+        _traced->to(_device);
+        _traced->eval();
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    void TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::clear_mllib(const APIData &ad) 
+    {
+
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    int TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::train(const APIData &ad, APIData &out) 
+    {
+        
+    }
+
+    template <class TInputConnectorStrategy, class TOutputConnectorStrategy, class TMLModel>
+    int TorchLib<TInputConnectorStrategy, TOutputConnectorStrategy, TMLModel>::predict(const APIData &ad, APIData &out) 
+    {
+        APIData params = ad.getobj("parameters");
+        APIData output_params = params.getobj("output");
+
+        TInputConnectorStrategy inputc(this->_inputc);
+        TOutputConnectorStrategy outputc;
+        try {
+            inputc.transform(ad);
+        } catch (...) {
+            throw;
+        }
+
+        Tensor output = _traced->forward({inputc._in.to(_device)}).toTensor();
```

Comment:
We want to try / catch around the forward call so that we can have a message within the `catch` scope and before the `throw`. 

This is because there are many memory and network related common errors, and we want to signify what the error is. For this reason we may want to `catch(std::exception &e)` report `e.what()` in the error message.